stellarphot/core.py

```diff
@@ -715,6 +800,225 @@ def catalog_name(self):
     def catalog_source(self):
         return self.meta['catalog_source']
 
+    @staticmethod
+    def _tidy_vizier_catalog(data, mag_column_regex, color_column_regex):
+        """
+        Transform a Vizier catalog with magnitudes into tidy structure. Or
+        at least tidier -- this only handles changing magnitude and color
+        columns into tidy format.
+
+        Parameters
+        ----------
+
+        data : `astropy.table.Table`
+            Table of catalog information. There are no restrictions on the columns.
+
+        mag_column_regex : str
+            Regular expression to match magnitude columns.
+
+        color_column_regex : str
+            Regular expression to match color columns.
+
+        Returns
+        -------
+
+        `astropy.table.Table`
+            Table with magnitude and color columns in tidy format. All other
+            columns are preserved as they were in the input data.
+        """
+
+        mag_re = re.compile(mag_column_regex)
+        color_re = re.compile(color_column_regex)
+
+        # Find all the magnitude and color columns
+        mag_match = [mag_re.match(col) for col in data.colnames]
+        color_match = [color_re.match(col) for col in data.colnames]
+
+        # create a single list of all the matches
+        matches = [m_match if m_match else c_match
+                   for m_match, c_match in zip(mag_match, color_match)]
```

Comment:
To you identify the magnitude and color columns and then build a list of matches that consists of the magnitude column name (when available) otherwise a color column name.  Why?  These are column names, if there are 3 magnitudes and 4 color columns, why would you only keep the last color column (ignoring the first three color columns since there are three magnitude columns already).

Reply:
My intent is to capture all of the mag and color columns -- I'll write a test case that checks that that really happens.

Each individual column is either a color column (e.g. name `g-r`) or a magnitude (e.g. `g_mag` or `r_mag`), so for a set of columns named `['RA', 'DEC', 'g_mag', 'r_mag', 'g-r'] the two match lists should be

```python
mag_match =  [None, None, <Match object>, <Match object>,                   None]
color_match = [None, None,                   None,                   None, <Match object>]
```

so the list comprehension is grabbing whichever match is not `None`, if any.

Reply:
Got it.